datalad/support/exceptions.py

```diff
@@ -62,6 +67,10 @@ def __init__(self, exc=None, limit=None, capture_locals=False):
             )
             traceback.clear_frames(exc.__traceback__)
 
+        # log the captured exception
```

Comment:
I can only attach the comment here, but it really is about the optional nature of `exc` argument. Why is this done? Just because it is possible? What are the performance implications of `exc_info`? Under which conditions would I pass or not pass `exc`? My gut feeling is that `exc` should be a mandatory parameter, and exceptions should be caught with something like `Exception as e` and pass in, instead of promoting (double) lookups.

Reply:
TBH, that's a remainder from previous iteration with the purpose of being "compatible" with old `exc_str` behavior. I do agree, though, and I'm happy to remove it.

Reply:
According to git grep there's only one spot, where `exc_str` is called without argument. So - will adjust that spot and make it mandatory.